crates/autopilot/src/run_loop.rs

```diff
@@ -310,6 +311,24 @@ impl RunLoop {
 
     /// Runs the solver competition, making all configured drivers participate.
     async fn competition(&self, id: AuctionId, auction: &Auction) -> Vec<Participant<'_>> {
+        // Upload the auction to S3 in the background.
+        if let Some(uploader) = self.uploader.clone() {
+            let auction = auction.clone();
```

Comment:
nit: Could be a separate function.

Reply:
Let's move this to `infra` as per https://github.com/cowprotocol/services/pull/2206